tcms/testcases/tests/test_views.py

```diff
@@ -228,6 +232,95 @@ def _assertTestCase(self, test_case):
         self.assertEqual(test_case.notes, self.notes)
 
 
+class TestNewCasePermission(PermissionsTestCase):
+    permission_label = 'testcases.add_testcase'
+    http_method_names = ['get', 'post']
+    url = reverse('testcases-new')
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.tester = UserFactory()
+        cls.case_status_confirmed = TestCaseStatus.objects.get(name='CONFIRMED')
+        cls.case_status_proposed = TestCaseStatus.objects.get(name='PROPOSED')
```

Comment:
Use methods get_confirmed/get_proposed in the TestCaseStatus class and make sure you are only creating data which is necessary. If it is not used then it's not needed.

Reply:
I do need the 'case_status' parameter in cls.post_data while POSTing because it is a required field. And for that I need cls.case_status_confirmed. I have removed cls.case_status_proposed

Reply:
Not using get_confirmed method like shown previously.